Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/EditorComponentBase.cpp

```diff
@@ -49,6 +49,17 @@ namespace AzToolsFramework
             m_transform = nullptr;
         }
 
+        void EditorComponentBase::SetEntity(AZ::Entity* entity)
+        {
+            Component::SetEntity(entity);
+
+            if (m_alias.empty())
+            {
+                AZ_Assert(GetId() != AZ::InvalidComponentId, "ComponentId is invalid.");
```

Comment:
Component::SetEntity() and Entity::AddComponent() are not guaranteed to have a valid m_id by the time code reaches this point. So, instead of asserting here, just check that m_id is valid and only create alias if it is.

Reply:
Oh yeah it looks like m_id can be invalid if SetEntity is called with null. I'll keep the assert but add a check if entity is valid before setting the alias. m_id is expected to be valid if the component is on a valid entity.